File: C# Basics/NestedConditions/FishingBoat.cs
using System;
using System.Runtime.ExceptionServices;

namespace FishingBoat
{
    class Program
    {
        static void Main(string[] args)
        {
            double budget = double.Parse(Console.ReadLine());
            string season = Console.ReadLine();
            int fishermen = int.Parse(Console.ReadLine());

            double boatPrice = 0;

            switch (season)
            {
                case "Spring":
                    boatPrice = 3000;
                    break;
                case "Summer":
                case "Autumn":
                    boatPrice = 4200;
                    break;
                case "Winter":
                    boatPrice = 2600;
                    break;
            }

                if (fishermen <= 6)
                {
                    boatPrice *= (1 - 10.0 / 100);
                }
                else if (fishermen >= 7 && fishermen <= 11)
                {
                    boatPrice *= (1 - 15.0 / 100);
                }
                else if (fishermen >= 12)
                {
                    boatPrice *= (1 - 25.0 / 100);
                }

            if ((season == "Spring" || season == "Summer" || season == "Winter") && fishermen % 2 == 0)
            {
                boatPrice *= (1 - 5.0 / 100);
            }

            double difference = budget - boatPrice;
            string output = string.Empty;

            if (budget >= boatPrice)
            {
                output = $"Yes! You have {difference:f2} leva left.";
            }
            else
            {
                output = $"Not enough money! You need {Math.Abs(difference):f2} leva.";
            }

            Console.WriteLine(output);
        }
    }
}

File: Java Basics/ConditionsAdvancedEx/FishingBoat.java
package AdvancedConditionsEx;

import java.util.Scanner;

public class FishingBoat {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        double budget = Double.parseDouble(scanner.nextLine());
        String season = scanner.nextLine();
        int fishermen = Integer.parseInt(scanner.nextLine());

        double boatPrice = 0;

        switch (season) {
            case "Spring":
                boatPrice = 3000;
                break;
            case "Summer":
            case "Autumn":
                boatPrice = 4200;
                break;
            case "Winter":
                boatPrice = 2600;
                break;
        }

        if (fishermen <= 6) {
            boatPrice *= (1 - 10 / 100d);
        } else if (fishermen <= 11) {
            boatPrice *= (1 - 15 / 100d);
        } else {
            boatPrice *= (1 - 25 / 100d);
        }

        if (fishermen % 2 == 0 && !season.equals("Autumn")) {
            boatPrice *= (1 - 5 / 100d);
        }

        if (budget >= boatPrice) {
            System.out.printf("Yes! You have %.2f leva left.", budget - boatPrice);
        } else {
            System.out.printf("Not enough money! You need %.2f leva.", boatPrice - budget);
        }
    }
}
